openerp/openupgrade/doc/source/development.rst

```diff
@@ -1,7 +1,34 @@
+=========================================================
 Migration script development
-++++++++++++++++++++++++++++
+=========================================================
 
 .. toctree::
    :maxdepth: 2
 
    devfaq
+
+Migration Scrips consist of two (or more) main files in every module's migration directory located in the target source code:
+
+* pre-migration.py (or pre-anythingyouwant.py)
+* post-migration.py (or post-anythingyouwant.py)
+
+What they do is derived from their names, you should check the API Documentation_ of the openupgradelib_ for details of library methods that are readily available for your migration task.
+
+The general purpose of those scripts is to manipulate directly the database entries of the source database in order to change them in a way so that they are compatible with the target database source code. To make that happen smoothly, is your task.
+When you run (TODO DOC HOWTO RUN A CUSTOM MIGRATION) a migration, Odoo's *internal* updating mechanism will be used (OpenUpgrade is providing sugar code for the migration task only). (@Experts: Under what circumstances a failing migration is reverted?)
+
+General recommendations about 'pre-migration.py' and 'post-migration.py'
+------------------------------------------------------------------------
+tbd by an expert
+
+Special Case: Module Renames and Module Merges
+----------------------------------------------
+tbd by an expert
```

Comment:
Thanks, well hm, it is still very confusing, as the software does not behave as you said, in `apriori.py`, we have:  `'l10n_co_base': 'l10n_co_geo',` ` l10n_co_base` being the old name. A breakpoint in the openupgrade server `migrations.py` in the `migrate_module` method AFTER the `hasattr(pkg, 'update')` statment is never hit when launching with `--update l10n_co_geo`, neither so with `--update l10n_co_base`. So probably your statement about the `apriori.py` considerations at migrate time does have any implicit limitations? It seems that on one of the eligible modules, the script doesn't even enter the `migrate_module` method at all. It just returns without any error specific, nothing... Do you have an idea?